lib/media_server_web/controllers/auth_controller.ex

```diff
@@ -0,0 +1,57 @@
+defmodule MediaServerWeb.AuthController do
+  use MediaServerWeb, :controller
+
+  @doc """
+  This action is reached via `/auth/:provider` and redirects to the OAuth2 provider
+  based on the chosen strategy.
+  """
+  def index(conn, %{"provider" => provider}) do
+    redirect conn, external: authorize_url!(provider)
+  end
+
+  def delete(conn, _params) do
+    conn
+    |> put_flash(:info, "You have been logged out!")
+    |> configure_session(drop: true)
+    |> redirect(to: "/")
+  end
+
+  @doc """
+  This action is reached via `/auth/:provider/callback` is the callback URL that
+  the OAuth2 provider will redirect the user back to with a `code` that will
+  be used to request an access token. The access token will then be used to
+  access protected resources on behalf of the user.
+  """
+  def callback(conn, %{"provider" => provider, "code" => code}) do
+    # Exchange an auth code for an access token
+    client = get_token!(provider, code)
+
+    # Request the user's data with the access token
+    if user = MediaServer.Accounts.get_user_by_email(get_user!(provider, client).email) do
+
+      MediaServerWeb.UserAuth.log_in_user(conn, user)
+    end
```

Comment:
@onedr0p

I would like some feedback on this. Here it gets a user registered in the Midarr database that matches the email returned by Authentik, then logs them in.

What's your take on this, and the auth flow?

If a user returned by Authentik doesn't exist in Midarr - register them, then log them in?

Reply:
I wouldn't auto-create users since the source of truth is the OIDC provider, meaning to say they would need to be added to the OIDC provider first.

Reply:
I was thinking oauth / OIDC provider would be an optional alternative, with Midarr still the source of truth for user auth. There would need to be some level of user mapping to support both options.

Reply:
For now just to keep moving forward - I'll go with the logic as above (user must exist in Midarr and OIDC provider to login).

We can continue to iterate on the flow in future iterations 😄

Reply:
I meant to comment and say pretty much that. 😅